ecosystem/node-checker/src/server/generate_openapi.rs

```diff
@@ -1,16 +1,64 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
-#![allow(unused_variables)]
-
+use crate::{metric_collector::ReqwestMetricCollector, runner::BlockingRunner};
 use anyhow::Result;
-use clap::Parser;
+use clap::{ArgEnum, Parser};
+use std::{collections::HashMap, path::PathBuf};
+use url::Url;
+
+use super::{
+    api::{build_openapi_service, Api},
+    configurations_manager::{ConfigurationsManager, NodeConfigurationWrapper},
+};
+
+#[derive(ArgEnum, Clone, Debug)]
+enum OutputFormat {
+    Json,
+    Yaml,
+}
 
 #[derive(Clone, Debug, Parser)]
-pub struct GenerateOpenapi {}
+pub struct GenerateOpenapi {
+    /// What address to listen on.
+    #[clap(long, default_value = "http://0.0.0.0")]
+    pub listen_address: Url,
+
+    /// What port to listen on.
+    #[clap(long, default_value = "20121")]
+    pub listen_port: u16,
+
+    #[clap(short, long)]
+    output_path: Option<PathBuf>,
```

Comment:
nit: maybe add a comment that if this is not provided it will be printed to the console?